### PR TITLE
Use GetRuntimeDir to setup auth.json for login

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@mtrmac Pointed out that we have code duplication in libpod to get rootless runtimedir.  Will use the pkg/util code rather then using the dups.

@QiWang19 PTAL

**Comment by QiWang19** [comment]
LGTM

File: cmd/podman/shared/funcs.go [hunk 1]
```diff
@@ -6,20 +6,22 @@ import (
...
+
 	authfile = os.Getenv("REGISTRY_AUTH_FILE")
 	if authfile != "" {
 		return authfile
 	}
-	runtimeDir := os.Getenv("XDG_RUNTIME_DIR")
-	if runtimeDir != "" {
-		return filepath.Join(runtimeDir, "containers/auth.json")
+
+	if runtimeDir, err := util.GetRootlessRuntimeDir(); err == nil {
```
Comment (mtrmac) [inline cmd/podman/shared/funcs.go]: The comment to that function says
```go
// GetRootlessRuntimeDir returns the runtime directory when running as non root
```
and callers in `libpod/runtime.go` are all guarded by `rootless.IsRootless()`.  OTOH `libpod/oci*.go` seems to call it without that guard.

I don’t know which one is right, but clarifying that one way or the other would be nice.

Comment (rhatdan) [reply cmd/podman/shared/funcs.go]: We should probably make this the GetRuntimeDir function.  It seems to return the right thing.  I tried this out in a bunch of different ways including as root.

Comment (rhatdan) [reply cmd/podman/shared/funcs.go]: @giuseppe WDYT?

Comment (giuseppe) [reply cmd/podman/shared/funcs.go]: yes, I think we can rename it to `GetRuntimeDir()` given it works also as root

**Comment by mheon** [comment]
Changes LGTM

File: cmd/podman/shared/funcs.go [hunk 2]
```diff
@@ -6,20 +6,22 @@ import (
...
+
 	authfile = os.Getenv("REGISTRY_AUTH_FILE")
 	if authfile != "" {
 		return authfile
 	}
-	runtimeDir := os.Getenv("XDG_RUNTIME_DIR")
-	if runtimeDir != "" {
-		return filepath.Join(runtimeDir, "containers/auth.json")
+
+	if runtimeDir, err := util.GetRuntimeDir(); err == nil {
```
Comment (mtrmac) [inline cmd/podman/shared/funcs.go]: Is it OK to just silently ignore the error here?

Comment (rhatdan) [reply cmd/podman/shared/funcs.go]: Yes I think we can.

**Pull request merged**